crates/ruff_cli/src/diagnostics.rs

```diff
@@ -135,9 +151,14 @@ pub(crate) fn lint_path(
             .iter_enabled()
             .any(|rule_code| rule_code.lint_source().is_pyproject_toml())
         {
-            let contents = std::fs::read_to_string(path)?;
+            let contents = match std::fs::read_to_string(path) {
+                Ok(contents) => contents,
+                Err(err) => {
+                    return Ok(io_error_diagnostics(err, path));
+                }
+            };
             let source_file = SourceFileBuilder::new(path.to_string_lossy(), contents).finish();
-            lint_pyproject_toml(source_file, &settings.lib)?
```

Comment:
Are these not already treated as `IOError` via the `IOError` handling in `run.rs`? Like this block:

```rust
error!(
    "{}{}{} {message}",
    "Failed to lint ".bold(),
    fs::relativize_path(path).bold(),
    ":".bold()
);
let settings = resolver.resolve(path, pyproject_config);
if settings.rules.enabled(Rule::IOError) {
    let file =
        SourceFileBuilder::new(path.to_string_lossy().as_ref(), "").finish();

    Diagnostics::new(
        vec![Message::from_diagnostic(
            Diagnostic::new(IOError { message }, TextRange::default()),
            file,
            TextSize::default(),
        )],
        ImportMap::default(),
    )
} else {
    Diagnostics::default()
}
```

Reply:
With this change, do we not lose that warning message that appears separately from the diagnostics?